wallet/txrules/rules.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2016 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+// Package txrules provides transaction rules that should be followed by
+// transaction authors for wide mempool acceptance and quick mining.
+package txrules
+
+import (
+	"errors"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+)
+
+// DefaultRelayFeePerKb is the default minimum relay fee policy for a mempool.
+const DefaultRelayFeePerKb btcutil.Amount = 1e3
+
+// IsDust determines whether a transaction output value and script length would
+// cause the output to be considered dust.  Transactions with dust outputs are
+// not standard and are rejected by mempools with default policies.
+func IsDust(amount btcutil.Amount, scriptSize int, relayFeePerKb btcutil.Amount) bool {
+	// Calculate the total (estimated) cost to the network.  This is
+	// calculated using the serialize size of the output plus the serial
+	// size of a transaction input which redeems it.  The output is assumed
+	// to be compressed P2PKH as this is the most common script type.  Use
+	// the average size of a compressed P2PKH redeem input (148) rather than
+	// the largest possible (txsizes.RedeemP2PKHInputSize).
+	totalSize := 8 + wire.VarIntSerializeSize(uint64(scriptSize)) +
+		scriptSize + 148
+
+	// Dust is defined as an output value where the total cost to the network
+	// (output size + input size) is greater than 1/3 of the relay fee.
+	return int64(amount)*1000/(3*int64(totalSize)) < int64(relayFeePerKb)
+}
+
+// OutputIsDust determines whether a transaction output value is considered
+// dust.  Transactions with dust outputs are not standard and are rejected by
+// mempools with default policies.
+func OutputIsDust(output *wire.TxOut, relayFeePerKb btcutil.Amount) bool {
```

Comment:
I'll rename IsDust to IsDustAmount, OutputIsDust to IsDustOutput (for consistency), and check for unspendable outputs in IsDustOutput.